engine/common/rpc/convert/executor_metadata.go

```diff
@@ -0,0 +1,29 @@
+package convert
+
+import (
+	"github.com/onflow/flow/protobuf/go/flow/entities"
+
+	"github.com/onflow/flow-go/model/access"
+)
+
+func ExecutorMetadataToMessage(metadata *access.ExecutorMetadata) (*entities.ExecutorMetadata, error) {
```

Comment:
we're already passing `access.ExecutorMetadata`  by value everywhere. if you also pass it by value here, then the nil check isn't required